inconspiquous/tools/quopt_main.py

```diff
@@ -1,25 +1,66 @@
-from inconspiquous.dialects import get_all_dialects
-from inconspiquous.transforms import get_all_passes
+from __future__ import annotations
+
+from typing import IO
 from xdsl.xdsl_opt_main import xDSLOptMain
+from xdsl.dialects.builtin import ModuleOp
+
+from inconspiquous.backend.qir import pyqir_available
 
 
 class QuoptMain(xDSLOptMain):
+    """Main entry point for quantum optimization passes"""
+
     def register_all_dialects(self):
-        for name, dialect in get_all_dialects().items():
-            self.ctx.register_dialect(name, dialect)
+        """Register all dialects"""
+        super().register_all_dialects()
+        # Add quantum-specific dialects here
 
     def register_all_passes(self):
-        for name, pass_ in get_all_passes().items():
-            self.register_pass(name, pass_)
+        """Register all passes"""
+        super().register_all_passes()
+        # Add quantum-specific passes here
 
     def register_all_targets(self):
         super().register_all_targets()
+        try:
+            from inconspiquous.backend.qir import emit_qir_module
```

Comment:
Can you please move this into the function? And then handle the import error there, with a helpful error message